### ISSUE TITLE
Schema Type Inference

### ISSUE THREAD

**Issue by DrewRidley** [siteforge-io/surreal-codegen#2]
Hi!

First of all, thank you for your work developing this type generator! It makes it possible to catch errors across a large codebase much easier, especially in CI/CD workflows.

The documentation mentions some examples where you can 'cast' the provided variables to the desired type. This is reflected upstream in the typescript definitions. In my experienced with a rust based type-generator, I found it is possible within ``UPDATE``, ``CREATE`` and other statements, to infer the type at compile time.

For example if I did ``UPDATE $user SET email = $email``, it can infer the required type of email as a string, because the field email is defined as a string in the schema.

This would reduce a lot of boilerplate while maintaining the same if not a higher safety margin on static typing.

Let me know your thoughts, and thanks again for your work on this project.

**Comment by AlbertMarashi** [comment]
Yes, that feature is definitely on the roadmap as noted in the `README.md` and definitely something we want to work on, as it is very important.

We'll try to prioritize it soon.

**Comment by AlbertMarashi** [comment]
FYI, still have plans to work on this at some point soon, because it would be a huge win for reducing boilerplate, but most importantly, it actually prevents developer mistyping, because the developer could actually specify an invalid type, and currently that is **not checked**
